File: domain/src/main/java/com/beat/domain/performanceimage/repository/PerformanceImageRepository.java
package com.beat.domain.performanceimage.repository;

import java.util.List;
import java.util.Optional;

import com.beat.domain.performanceimage.domain.PerformanceImage;

public interface PerformanceImageRepository {

	Optional<PerformanceImage> findById(Long id);

	List<PerformanceImage> findAllByPerformanceId(Long performanceId);

	List<Long> findIdsByPerformanceId(Long performanceId);

	PerformanceImage save(PerformanceImage performanceImage);

	List<PerformanceImage> saveAll(List<PerformanceImage> performanceImages);

	void delete(PerformanceImage performanceImage);

	void deleteByPerformanceId(Long performanceId);
}

File: domain/src/main/kotlin/com/beat/domain/performanceimage/repository/PerformanceImageRepository.kt
package com.beat.domain.performanceimage.repository

import com.beat.domain.performanceimage.domain.PerformanceImage
import java.util.*

@JvmSuppressWildcards
interface PerformanceImageRepository {
    fun findById(id: Long?): Optional<PerformanceImage>

    fun findAllByPerformanceId(performanceId: Long?): List<PerformanceImage>

    fun findIdsByPerformanceId(performanceId: Long?): List<Long>

    fun save(performanceImage: PerformanceImage): PerformanceImage

    fun saveAll(performanceImages: List<PerformanceImage>): List<PerformanceImage>

    fun delete(performanceImage: PerformanceImage)

    fun deleteByPerformanceId(performanceId: Long?)
}
